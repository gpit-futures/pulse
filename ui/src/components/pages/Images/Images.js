import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-bootstrap';
import classNames from 'classnames';
import _ from 'lodash/fp';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { lifecycle, compose } from 'recompose';

import PluginListHeader from '../../plugin-page-component/PluginListHeader';
import PluginMainPanel from '../../plugin-page-component/PluginMainPanel';
import { columnsConfig, defaultColumnsSelected } from './table-columns.config'
import { valuesNames } from './forms.config';
import { fetchPatientImagesRequest } from './ducks/fetch-patient-images.duck';
import { fetchPatientImagesDetailRequest } from './ducks/fetch-patient-images-detail.duck';
import { fetchSeriesRequest } from './ducks/fetch-all-series.duck';
import { fetchSeriesDetailRequest } from './ducks/fetch-series-detail.duck';
import { fetchPatientImagesOnMount, fetchSeriesOnMount } from '../../../utils/HOCs/fetch-patients.utils';
import { patientImagesSelector, seriesDetailAndInstanceIdsSelector, allSeriesSelector } from './selectors';
import { clientUrls } from '../../../config/client-urls.constants';
import ImagesDetail from './ImagesDetail/ImagesDetail';
import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';
import { operationsOnCollection } from '../../../utils/plugin-helpers.utils';

const IMAGES_MAIN = 'imagesMain';
const IMAGES_DETAIL = 'imagesDetail';
const IMAGES_PANEL = 'imagesPanel';
const IMAGES_DETAIL_PANEL = 'imagesDetailPanel';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ fetchPatientImagesRequest, fetchPatientImagesDetailRequest, fetchSeriesRequest, fetchSeriesDetailRequest }, dispatch) });

@connect(patientImagesSelector, mapDispatchToProps)
@connect(seriesDetailAndInstanceIdsSelector)
@connect(allSeriesSelector)
@compose(lifecycle(fetchPatientImagesOnMount), lifecycle(fetchSeriesOnMount))
export default class Images extends PureComponent {
  static propTypes = {
    allImages: PropTypes.arrayOf(PropTypes.object),
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  state = {
    nameShouldInclude: '',
    selectedColumns: defaultColumnsSelected,
    openedPanel: IMAGES_PANEL,
    columnNameSortBy: valuesNames.STUDY_DESCRIPTION,
    sortingOrder: 'asc',
    expandedPanel: 'all',
    isBtnCreateVisible: false,
    isBtnExpandVisible: false,
    isAllPanelsVisible: false,
    isDetailPanelVisible: false,
    isSecondPanel: false,
    editedPanel: {},
    offset: 0,
    isLoading: true,
  };

  componentWillReceiveProps() {
    const sourceId = this.context.router.route.match.params.sourceId;
    const userId = this.context.router.route.match.params.userId;

    //TODO should be implemented common function, and the state stored in the store Redux
    if (this.context.router.history.location.pathname === `${clientUrls.PATIENTS}/${userId}/${clientUrls.IMAGES}/${sourceId}` && sourceId !== undefined) {
      this.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true })
    }
    if (this.context.router.history.location.pathname === `${clientUrls.PATIENTS}/${userId}/${clientUrls.IMAGES}`) {
      this.setState({ isSecondPanel: false, isBtnExpandVisible: false, openedPanel: IMAGES_PANEL, isDetailPanelVisible: false, expandedPanel: 'all' })
    }

    /* istanbul ignore next */
    setTimeout(() => {
      this.setState({ isLoading: false })
    }, 500)
  }

  handleExpand = (name, currentPanel) => {
    if (currentPanel === IMAGES_MAIN) {
      if (this.state.expandedPanel === 'all') {
        this.setState({ expandedPanel: name });
      } else {
        this.setState({ expandedPanel: 'all' });
      }
    } else if (this.state.expandedPanel === 'all') {
      this.setState({ expandedPanel: name, openedPanel: name });
    } else {
      this.setState({ expandedPanel: 'all' });
    }
  };

  handleFilterChange = ({ target: { value } }) => this.setState({ nameShouldInclude: _.toLower(value) });

  handleHeaderCellClick = (e, { name, sortingOrder }) => this.setState({ columnNameSortBy: name, sortingOrder });

  handleDetailImagesClick = (sourceId) => {
    const { actions, userId } = this.props;
    this.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, openedPanel: IMAGES_PANEL, editedPanel: {}, expandedPanel: 'all', isLoading: true });
    actions.fetchSeriesRequest({ userId, studyId: sourceId, source: 'orthanc' });
    this.context.router.history.push(`${clientUrls.PATIENTS}/${userId}/${clientUrls.IMAGES}/${sourceId}`);
  };

  handleSetOffset = offset => this.setState({ offset });

  formToShowCollection = (collection) => {
    const { columnNameSortBy, sortingOrder, nameShouldInclude } = this.state;

    collection = operationsOnCollection.modificate(collection, [{
      keyFrom: valuesNames.DATE_RECORDED,
      keyTo: `${valuesNames.DATE_RECORDED}Convert`,
      fn: getDDMMMYYYY,
    }]);

    return operationsOnCollection.filterAndSort({
      collection,
      filterBy: nameShouldInclude,
      sortingByKey: columnNameSortBy,
      sortingByOrder: sortingOrder,
      filterKeys: [valuesNames.STUDY_DESCRIPTION, `${valuesNames.DATE_RECORDED}Convert`, valuesNames.SOURCE],
      modeSorting: {
        number: [valuesNames.DATE_RECORDED],
      },
    });
  };

  handleShow = (name) => {
    this.setState({ openedPanel: name })
  };

  render() {
    const { selectedColumns, columnNameSortBy, sortingOrder, isSecondPanel, isDetailPanelVisible, isBtnExpandVisible, expandedPanel, openedPanel, isBtnCreateVisible, editedPanel, offset, isLoading } = this.state;
    const { allImages, instanceIds, serieDetail, allSerie } = this.props;

    const isPanelDetails = (expandedPanel === IMAGES_DETAIL || expandedPanel === IMAGES_PANEL || expandedPanel === IMAGES_DETAIL_PANEL);
    const isPanelMain = (expandedPanel === IMAGES_MAIN);

    const columnsToShowConfig = columnsConfig.filter(columnConfig => selectedColumns[columnConfig.key]);

    const filteredImages = this.formToShowCollection(allImages);

    let sourceId;
    if (!_.isEmpty(serieDetail)) {
      sourceId = allSerie[valuesNames.STUDY_ID];
    }

    return (<section className="page-wrapper">
      <div className={classNames('section', { 'full-panel full-panel-main': isPanelMain, 'full-panel full-panel-details': isPanelDetails })}>
        <Row>
          {(isPanelMain || expandedPanel === 'all') ? <Col xs={12} className={classNames({ 'col-panel-main': isSecondPanel })}>
            <div className="panel panel-primary">
              <PluginListHeader
                onFilterChange={this.handleFilterChange}
                panelTitle="Images"
                isBtnExpandVisible={isBtnExpandVisible}
                name={IMAGES_MAIN}
                onExpand={this.handleExpand}
                currentPanel={IMAGES_MAIN}
              />
              <PluginMainPanel
                headers={columnsToShowConfig}
                resourceData={allImages}
                emptyDataMessage="No images"
                onHeaderCellClick={this.handleHeaderCellClick}
                onCellClick={this.handleDetailImagesClick}
                columnNameSortBy={columnNameSortBy}
                sortingOrder={sortingOrder}
                table="images"
                filteredData={filteredImages}
                totalEntriesAmount={_.size(filteredImages)}
                offset={offset}
                setOffset={this.handleSetOffset}
                isBtnCreateVisible={isBtnCreateVisible}
                onCreate={this.handleCreate}
                id={sourceId}
                isLoading={isLoading}
              />
            </div>
          </Col> : null }
          {(expandedPanel === 'all' || isPanelDetails) && isDetailPanelVisible ? <Col xs={12} className={classNames({ 'col-panel-details': isSecondPanel })}>
            <ImagesDetail
              onExpand={this.handleExpand}
              name={IMAGES_DETAIL}
              openedPanel={openedPanel}
              expandedPanel={expandedPanel}
              currentPanel={IMAGES_DETAIL}
              detail={serieDetail}
              onEdit={this.handleEdit}
              editedPanel={editedPanel}
              onShow={this.handleShow}
              instanceIds={instanceIds}
            />
          </Col> : null}
        </Row>
      </div>
    </section>)
  }
}
