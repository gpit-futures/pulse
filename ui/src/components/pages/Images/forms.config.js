export const valuesNames = {
  TYPE: 'noteType',
  NOTES: 'notes',
  AUTHOR: 'author',
  STUDY_DESCRIPTION: 'studyDescription',
  DATE_RECORDED: 'dateRecorded',

  MODALITY: 'modality',
  PROTOCOL_NAME: 'protocolName',
  STATION_NAME: 'stationName',
  OPERATORS_NAME: 'operatorsName',
  SERIES_DATE: 'seriesDate',
  SERIES_TIME: 'seriesTime',
  SERIES_NUMBER: 'seriesNumber',

  SOURCE: 'source',
  STUDY_ID: 'studyId',
  SOURCE_ID: 'sourceId',
};

export const valuesLabels = {
  TYPE: 'Type',
  NOTES: 'Note',
  AUTHOR: 'Author',
  DATE: 'Date',

  MODALITY: 'Modality',
  PROTOCOL_NAME: 'Protocol Name',
  STATION_NAME: 'Station Name',
  OPERATORS_NAME: 'Operator Name',
  SERIES_DATE: 'Series Date',
  SERIES_TIME: 'Series Time',
  SERIES_NUMBER: 'Series No:',

  SOURCE: 'Source',
  SOURCE_ID: 'Source ID',
};
