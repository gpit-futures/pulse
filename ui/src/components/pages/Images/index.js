import { combineEpics } from 'redux-observable';

import asyncComponent from '../../../components/containers/AsyncComponent/AsyncComponent';
import { clientUrls } from '../../../config/client-urls.constants';

import { fetchPatientImagesEpic } from './ducks/fetch-patient-images.duck';
import { fetchPatientImagesDetailEpic } from './ducks/fetch-patient-images-detail.duck';
import { fetchSeriesEpic } from './ducks/fetch-all-series.duck';
import { fetchSeriesDetailEpic } from './ducks/fetch-series-detail.duck';

import patientsImages from './ducks/fetch-patient-images.duck';
import imagesDetail from './ducks/fetch-patient-images-detail.duck';
import allSeries from './ducks/fetch-all-series.duck';
import seriesDetail from './ducks/fetch-series-detail.duck';

const epics = combineEpics(fetchPatientImagesEpic, fetchPatientImagesDetailEpic, fetchSeriesEpic, fetchSeriesDetailEpic);
const Images = asyncComponent(() => import(/* webpackChunkName: "images" */ './Images').then(module => module.default));

const reducers = {
  patientsImages,
  imagesDetail,
  allSeries,
  seriesDetail,
};

const sidebarConfig = { key: 'images', pathToTransition: '/images', name: 'Images', isVisible: true };

const routers = [
  { key: 'images', component: Images, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.IMAGES}` },
  { key: 'imagesDetail', component: Images, path: `${clientUrls.PATIENTS}/:userId/${clientUrls.IMAGES}/:sourceId` },
];

export default {
  component: Images,
  epics, reducers, sidebarConfig, routers,
}
