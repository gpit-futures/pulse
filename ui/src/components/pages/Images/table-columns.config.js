import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';

export const columnsConfig = [
  { key: 'studyDescription', title: 'Name', width: '56%' },
  { key: 'dateRecorded', title: 'Date', transformer: getDDMMMYYYY, width: '25%' },
  { key: 'source', title: 'Source', width: '19%' },
  { key: 'sourceId', title: 'sourceId', width: 0, display: 'none' },
];

export const defaultColumnsSelected = {
  studyDescription: true,
  dateRecorded: true,
  source: true,
  sourceId: true,
};
