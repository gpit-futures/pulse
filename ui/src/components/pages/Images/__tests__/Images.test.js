import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import Images from '../Images';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

// frequently used variables
const userId = '9999999000';
const sourceId = 'acb0eaf2-d1df-4c7a-9382-619b31935f2b';

// storage setup in different configurations
const mockStore = configureStore();
const storeResource = {
  patientsImages: {
    '9999999000': [
      {
        [valuesNames.STUDY_ID]: '55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba',
        [valuesNames.SOURCE]: 'orthanc',
        [valuesNames.STUDY_DESCRIPTION]: 'RT ANKLE',
        [valuesNames.DATE_RECORDED]: 736124400000,
        [valuesNames.STUDY_ID]: '55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba',
        dateRecordedConvert: '30-Apr-1993',
      },
    ],
  },
};
const storeEmpty = mockStore(Object.assign({
  seriesDetail: {
    '9999999000': {},
  },
  allSeries: {
    '9999999000': {},
  },
}, storeResource));
const storeWithDetail = mockStore(Object.assign({
  seriesDetail: {
    '9999999000': {
      [valuesNames.SOURCE]: 'orthanc',
      [valuesNames.SOURCE_ID]: '00d87cf2-13d39285-50905549-3faf78a7-59656dd3',
      [valuesNames.MODALITY]: 'CT',
      [valuesNames.SERIES_DATE]: 736124400000,
      [valuesNames.SERIES_TIME]: 37644000,
      [valuesNames.STATION_NAME]: 'CT01OC0',
      [valuesNames.OPERATORS_NAME]: 'Anonymized',
      [valuesNames.SERIES_NUMBER]: '365',
      instanceIds: [
        'd1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930',
      ],
      [valuesNames.PROTOCOL_NAME]: 'test',
    },
  },
  allSeries: {
    '9999999000': {
      [valuesNames.STUDY_ID]: '55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba',
      [valuesNames.SOURCE]: 'orthanc',
      seriesIds: [
        '9927f4d0-7ba0f736-ad0cd3a0-8cd74dba-5a23637e',
      ],
    },
  },
}, storeResource));

// configure context for various tests
const generateNewContext = (oldContext, pathname) => {
  const newContext = Object.assign({}, oldContext);
  newContext.router = Object.assign({}, newContext.router);
  newContext.router.history = Object.assign({}, newContext.router.history);
  newContext.router.history.location = { pathname };
  return newContext;
};
const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: `/patients/${userId}/images`,
      },
    },
    route: {
      match: {
        params: {
          sourceId,
          userId,
        },
      },
    },
  },
};
const contextDetail = generateNewContext(context, `/patients/${userId}/images/${sourceId}`);

const match = {
  params: {
    userId,
  },
};

describe('Component <Images />', () => {
  it('should renders correctly with seriesDetail and testing Detail Panel', () => {
    const component = shallow(
      <Images
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive()
      .dive()
      .dive();

    // Testing component handleDetailImagesClick methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.find('ImagesDetail')).toHaveLength(0);

    component.instance().handleDetailImagesClick('065d85e3-3cd5-4604-bb94-5685fffb193d');
    const componentStateAfterMethod = component.state();
    component.setState({ isSecondPanel: true, isDetailPanelVisible: true, isBtnExpandVisible: true, openedPanel: 'imagesPanel', editedPanel: {}, expandedPanel: 'all', isLoading: true });
    const componentStateAfterSetState = component.state();

    expect(componentStateAfterMethod).toEqual(componentStateAfterSetState);
    expect(component.find('ImagesDetail')).toHaveLength(1);
    expect(component).toMatchSnapshot();

    // Testing component handleShow methods
    expect(component.state().openedPanel).toEqual('imagesPanel');
    component.instance().handleShow('imagesDetailPanel');
    expect(component.state().openedPanel).toEqual('imagesDetailPanel');
    component.instance().handleShow('imagesPanel');
    expect(component.state().openedPanel).toEqual('imagesPanel');

    // Testing component handleExpand methods
    expect(component.find('PluginListHeader')).toHaveLength(1);
    expect(component.find('PluginMainPanel')).toHaveLength(1);
    expect(component.state().openedPanel).toEqual('imagesPanel');
    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('imagesPanel', 'imagesDetail');
    expect(component.state().openedPanel).toEqual('imagesPanel');
    expect(component.state().expandedPanel).toEqual('imagesPanel');
    component.setState({ openedPanel: 'imagesPanel', expandedPanel: 'imagesPanel' });
    expect(component.find('PluginListHeader')).toHaveLength(0);
    expect(component.find('PluginMainPanel')).toHaveLength(0);
    expect(component).toMatchSnapshot();

    component.setState({ openedPanel: 'imagesPanel', expandedPanel: 'all' });

    component.setState({ expandedPanel: 'all' });
    component.instance().handleExpand('imagesPanel', 'imagesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('imagesPanel', 'imagesMain');
    component.setState({ expandedPanel: 'test' });
    component.instance().handleExpand('imagesPanel', 'imagesTest');


    // Testing component detail form methods
    expect(component.state().editedPanel).toEqual({});

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing another methods', () => {
    const component = shallow(
      <Images
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive()
      .dive()
      .dive();

    // Testing component handleFilterChange methods
    expect(component.state().nameShouldInclude).toEqual('');
    component.instance().handleFilterChange({ target: { value: 'test' } });
    expect(component.state().nameShouldInclude).toEqual('test');
    component.instance().handleFilterChange({ target: { value: '' } });
    expect(component.state().nameShouldInclude).toEqual('');

    // Testing component handleHeaderCellClick methods
    expect(component.state().columnNameSortBy).toEqual('studyDescription');
    expect(component.state().sortingOrder).toEqual('asc');
    component.instance().handleHeaderCellClick({}, { name: 'dateRecorded', sortingOrder: 'desc' });
    expect(component.state().columnNameSortBy).toEqual('dateRecorded');
    expect(component.state().sortingOrder).toEqual('desc');

    // Testing component handleSetOffset methods
    expect(component.state().offset).toEqual(0);
    component.instance().handleSetOffset(10);
    expect(component.state().offset).toEqual(10);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly and testing context and lifecycle componentWillReceiveProps', () => {
    const component = shallow(
      <Images
        store={storeWithDetail}
        match={match}
      />, { context }).dive().dive().dive()
      .dive()
      .dive();

    component.setProps({ test: 'testing context' });
    component.setContext(contextDetail);
    component.setProps({ test: 'testing edit context' });

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly with empty store', () => {
    const component = shallow(
      <Images
        store={storeEmpty}
        match={match}
      />, { context }).dive().dive().dive()
      .dive()
      .dive();

    expect(component).toMatchSnapshot();
  });
});
