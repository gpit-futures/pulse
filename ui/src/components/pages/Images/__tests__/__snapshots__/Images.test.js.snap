// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Images /> should renders correctly and testing another methods 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="imagesMain"
            isBtnExpandVisible={false}
            name="imagesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Images"
          />
          <PluginMainPanel
            columnNameSortBy="studyDescription"
            emptyDataMessage="No images"
            filteredData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            headers={
              Array [
                Object {
                  "key": "studyDescription",
                  "title": "Name",
                  "width": "56%",
                },
                Object {
                  "key": "dateRecorded",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            id="55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba"
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onHeaderCellClick={[Function]}
            resourceData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            setOffset={[Function]}
            sortingOrder="asc"
            table="images"
            totalEntriesAmount={1}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Images /> should renders correctly and testing context and lifecycle componentWillReceiveProps 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="imagesMain"
            isBtnExpandVisible={true}
            name="imagesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Images"
          />
          <PluginMainPanel
            columnNameSortBy="studyDescription"
            emptyDataMessage="No images"
            filteredData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            headers={
              Array [
                Object {
                  "key": "studyDescription",
                  "title": "Name",
                  "width": "56%",
                },
                Object {
                  "key": "dateRecorded",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            id="55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba"
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onHeaderCellClick={[Function]}
            resourceData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            setOffset={[Function]}
            sortingOrder="asc"
            table="images"
            totalEntriesAmount={1}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ImagesDetail
          currentPanel="imagesDetail"
          detail={
            Object {
              "instanceIds": Array [
                "d1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930",
              ],
              "modality": "CT",
              "operatorsName": "Anonymized",
              "protocolName": "test",
              "seriesDate": 736124400000,
              "seriesNumber": "365",
              "seriesTime": 37644000,
              "source": "orthanc",
              "sourceId": "00d87cf2-13d39285-50905549-3faf78a7-59656dd3",
              "stationName": "CT01OC0",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          instanceIds={
            Array [
              "d1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930",
            ]
          }
          name="imagesDetail"
          onExpand={[Function]}
          onShow={[Function]}
          openedPanel="imagesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Images /> should renders correctly with empty store 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className=""
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="imagesMain"
            isBtnExpandVisible={false}
            name="imagesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Images"
          />
          <PluginMainPanel
            columnNameSortBy="studyDescription"
            emptyDataMessage="No images"
            filteredData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            headers={
              Array [
                Object {
                  "key": "studyDescription",
                  "title": "Name",
                  "width": "56%",
                },
                Object {
                  "key": "dateRecorded",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onHeaderCellClick={[Function]}
            resourceData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            setOffset={[Function]}
            sortingOrder="asc"
            table="images"
            totalEntriesAmount={1}
          />
        </div>
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Images /> should renders correctly with seriesDetail and testing Detail Panel 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-main"
        componentClass="div"
        xs={12}
      >
        <div
          className="panel panel-primary"
        >
          <PluginListHeader
            currentPanel="imagesMain"
            isBtnExpandVisible={true}
            name="imagesMain"
            onExpand={[Function]}
            onFilterChange={[Function]}
            panelTitle="Images"
          />
          <PluginMainPanel
            columnNameSortBy="studyDescription"
            emptyDataMessage="No images"
            filteredData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            headers={
              Array [
                Object {
                  "key": "studyDescription",
                  "title": "Name",
                  "width": "56%",
                },
                Object {
                  "key": "dateRecorded",
                  "title": "Date",
                  "transformer": [Function],
                  "width": "25%",
                },
                Object {
                  "key": "source",
                  "title": "Source",
                  "width": "19%",
                },
                Object {
                  "display": "none",
                  "key": "sourceId",
                  "title": "sourceId",
                  "width": 0,
                },
              ]
            }
            id="55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba"
            isBtnCreateVisible={false}
            isLoading={true}
            listPerPageAmount={10}
            offset={0}
            onCellClick={[Function]}
            onHeaderCellClick={[Function]}
            resourceData={
              Array [
                Object {
                  "dateRecorded": 736124400000,
                  "dateRecordedConvert": "30-Apr-1993",
                  "source": "orthanc",
                  "sourceId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                  "studyDescription": "RT ANKLE",
                  "studyId": "55a9fcd2-e8197ca2-1af7a8e2-0e1ab147-841c65ba",
                },
              ]
            }
            setOffset={[Function]}
            sortingOrder="asc"
            table="images"
            totalEntriesAmount={1}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ImagesDetail
          currentPanel="imagesDetail"
          detail={
            Object {
              "instanceIds": Array [
                "d1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930",
              ],
              "modality": "CT",
              "operatorsName": "Anonymized",
              "protocolName": "test",
              "seriesDate": 736124400000,
              "seriesNumber": "365",
              "seriesTime": 37644000,
              "source": "orthanc",
              "sourceId": "00d87cf2-13d39285-50905549-3faf78a7-59656dd3",
              "stationName": "CT01OC0",
            }
          }
          editedPanel={Object {}}
          expandedPanel="all"
          instanceIds={
            Array [
              "d1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930",
            ]
          }
          name="imagesDetail"
          onExpand={[Function]}
          onShow={[Function]}
          openedPanel="imagesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Images /> should renders correctly with seriesDetail and testing Detail Panel 2`] = `
<section
  className="page-wrapper"
>
  <div
    className="section full-panel full-panel-details"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="col-panel-details"
        componentClass="div"
        xs={12}
      >
        <ImagesDetail
          currentPanel="imagesDetail"
          detail={
            Object {
              "instanceIds": Array [
                "d1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930",
              ],
              "modality": "CT",
              "operatorsName": "Anonymized",
              "protocolName": "test",
              "seriesDate": 736124400000,
              "seriesNumber": "365",
              "seriesTime": 37644000,
              "source": "orthanc",
              "sourceId": "00d87cf2-13d39285-50905549-3faf78a7-59656dd3",
              "stationName": "CT01OC0",
            }
          }
          editedPanel={Object {}}
          expandedPanel="imagesPanel"
          instanceIds={
            Array [
              "d1d4875a-2a8267c2-2ee4f8f0-3d9e917e-9a141930",
            ]
          }
          name="imagesDetail"
          onExpand={[Function]}
          onShow={[Function]}
          openedPanel="imagesPanel"
        />
      </Col>
    </Row>
  </div>
</section>
`;

exports[`Component <Images /> should renders correctly with seriesDetail and testing Detail Panel 3`] = `
<section
  className="page-wrapper"
>
  <div
    className="section"
  >
    <Row
      bsClass="row"
      componentClass="div"
    />
  </div>
</section>
`;
